src/java.base/share/classes/java/security/Security.java

```diff
@@ -192,6 +199,22 @@ private static void initialize() {
             }
         }
 
+        // check if CRIU checkpoint mode is enabled, if it is - reconfigure the security providers
+        boolean isCRIUCapable = Boolean.valueOf(System.getProperty("org.eclipse.openj9.criu.isCRIUCapable", "false"));
+        if (isCRIUCapable) {
```

Comment:
this should use `InternalCRIUSupport.isCheckpointAllowed` instead

Reply:
@JasonFengJ9 Does the JPP pre-processor work for files in closedj9?

Reply:
yes, those files are overlayed on top of the openj9 files.

Reply:
The problem here is that it cant find the internal CRIU clases, although the should exist if compiled with `--enable-criu-support`, I had a similar issue here, https://github.com/eclipse-openj9/openj9/issues/14821

Reply:
Or maybe you mean the closed directory, not closedj9? The preprocessor doesn't work on the files in the closed directory, similarly to OpenJDK jcl code.

Reply:
> Does the JPP pre-processor work for files in closedj9?

Yeah, so a workaround is to move these changes to there.
Ideally extension repo can accept JPP as well.

Reply:
@ZainabF92 Can you build with https://github.com/eclipse-openj9/openj9/pull/14848 and see if that solves the issues you are seeing

Reply:
yeah that solves it

Reply:
I changed my code to use `InternalCRIUSupport.isCheckpointAllowed()` and `InternalCRIUSupport.isCRIUSupportEnabled()`